src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    protected a: number,
+    protected b: number,
+    protected c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const longestSide = Math.max(a, b, c);
+    const perimeter = a + b + c;
 
-export class Triangle implements Figure {}
+    if (longestSide >= perimeter - longestSide) {
+      throw new Error(
+        'The longest side must be shorter than the sum of the other two sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)) * 100,
+      ) / 100
+    );
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    protected radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius * this.radius * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle' = 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    protected width: number,
+    protected height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Width and height must be greater than 0');
```

Comment:
[CODE STYLE] - The error message can be improved by being more descriptive. Consider stating, 'Rectangle width and height must both be greater than 0.'